### PR TITLE
ssl-proxy: init at 0.2.7

### PR DESCRIPTION
**Pull request by konst-aa**
## Description of changes
Packaged ssl-proxy.
Homepage: https://github.com/suyashkumar/ssl-proxy
Description taken from the repo: [ssl-proxy is a] simple single-command SSL reverse proxy with autogenerated certificates (LetsEncrypt, self-signed).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
please use SRI hash for the fetcher (`hash = "...";` instead of `sha256 = "...";`), and we can also set `meta.mainProgram = "ssl-proxy";` here.

also, any reason to exclude darwin and windows? I see the original repo has support for those.

**Review by eclairevoyant — COMMENTED** [review]
also - typo, should be `maintainers` not `maintaners`

**Comment by konst-aa** [comment]
> also, any reason to exclude darwin and windows? I see the original repo has support for those.
Yeah, the only issue is that I only have access to a linux machine.

File: pkgs/tools/networking/ssl-proxy/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner = "suyashkumar";
+    repo = "ssl-proxy";
+    rev = "v${version}";
+    hash = "sha256-c9BLdDlkrg1z1QrO+vEAVyPtrV/nQcYlGXFmwfAOSpQ=";
+  };
+
+  vendorHash = "sha256-310K9ZSxy/OQ4HYFCcHQaj4NQwzATrOZ2YkhiSkhY5I=";
+
+  testPhase = ''
```
Comment (AndersonTorres) [inline pkgs/tools/networking/ssl-proxy/default.nix:15]: ??

Comment (konst-aa) [reply pkgs/tools/networking/ssl-proxy/default.nix:15]: What's the problem?

**Review by NotAShelf — APPROVED** [review]

File: pkgs/tools/networking/ssl-proxy/default.nix (hunk 1 above)
Comment (h7x4) [reply pkgs/tools/networking/ssl-proxy/default.nix:15]: There is no such thing as a `testPhase`. Maybe you meant `checkPhase`?

https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases

Comment (konst-aa) [reply pkgs/tools/networking/ssl-proxy/default.nix:15]: Oh, my bad.

File: pkgs/tools/networking/ssl-proxy/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchFromGitHub {
+    owner = "suyashkumar";
+    repo = "ssl-proxy";
+    rev = "v${version}";
+    hash = "sha256-c9BLdDlkrg1z1QrO+vEAVyPtrV/nQcYlGXFmwfAOSpQ=";
+  };
+
+  vendorHash = "sha256-310K9ZSxy/OQ4HYFCcHQaj4NQwzATrOZ2YkhiSkhY5I=";
+
+  doCheckPhase = true;
```
Comment (nagy) [inline pkgs/tools/networking/ssl-proxy/default.nix:15]: I think you mean `doCheck`. But that should be enabled for go modules by default.

Comment (konst-aa) [reply pkgs/tools/networking/ssl-proxy/default.nix]: Oh yeah, my bad. Programming by coincidence :pensive:.  

I found the line in the `buildGoModule` implementation: `doCheck = args.doCheck or true;` 

Going back to my code though, should I set `doCheck = true;` to be more explicit?

Comment (nagy) [reply pkgs/tools/networking/ssl-proxy/default.nix]: I dont see other go package do it, so I would say no.

File: pkgs/tools/networking/ssl-proxy/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      notebook or your team jenkins instance. ssl-proxy autogenerates SSL certs and proxies
+      HTTPS traffic to an existing HTTP server in a single command.
+    '';
+    license = licenses.mit;
+    mainProgram = "ssl-proxy";
+    maintainers = [ maintainers.konst-aa ];
+    platforms = platforms.linux ++ platforms.darwin ++ platforms.windows ;
+  };
+}
+
```
Comment (AndersonTorres) [inline pkgs/tools/networking/ssl-proxy/default.nix:31]:
```suggestion
```

File: pkgs/tools/networking/ssl-proxy/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ lib, buildGo118Module, fetchFromGitHub }:
+buildGo118Module rec {
```
Comment (AndersonTorres) [inline pkgs/tools/networking/ssl-proxy/default.nix:2]:
```suggestion

buildGo118Module rec {
```

File: pkgs/tools/networking/ssl-proxy/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  version = "0.2.7";
+  pname = "ssl-proxy";
```
Comment (AndersonTorres) [inline pkgs/tools/networking/ssl-proxy/default.nix:4]:
```suggestion
  pname = "ssl-proxy";
  version = "0.2.7";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
Ping me 2 days to merge

**Comment by konst-aa** [comment]
@AndersonTorres is it merging time?

**Pull request merged**